File: demo_oneactivity_multifragment/src/main/java/com/example/demo_oneactivity_multifragment/MainActivity.java
package com.example.demo_oneactivity_multifragment;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;

/**
 * 一个activity 多个fragment结构示例
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        init();
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if(null!=data) {
            User user = (User) data.getSerializableExtra("user");
            Toast.makeText(this,user.toString(),Toast.LENGTH_SHORT).show();
        }
    }

    private void init() {
        startActivityForResult(new Intent(this, LoginActivity.class), 1);
    }
}

File: demo_kotlin/src/main/java/cn/yt/demo_kotlin/MainActivity.kt
package cn.yt.demo_kotlin

import android.content.Context
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //fbc 控件设置值
        tv.text = "阿尼尼"
        tv.setOnClickListener {
//           Toast.makeText(this,"sdasd",Toast.LENGTH_SHORT).show()
//            toast("dasdad")  错误
        }
    }


    /**
     * 函数扩展
     *  fun 类型.函数(参数)
     */
    fun Context.toast(message:String, length:Int = Toast.LENGTH_LONG){
        Toast.makeText(this,message,length)
    }
}
